common/config/config.go

```diff
@@ -502,15 +510,21 @@ func (c *Config) fillDefaults() error {
 			c.Persistence.DataStores[k] = store
 		}
 	}
-	// filling RPCName with a default value if empty
-	if c.ClusterMetadata != nil {
-		for k, cluster := range c.ClusterMetadata.ClusterInformation {
-			if cluster.RPCName == "" {
-				cluster.RPCName = "cadence-frontend"
-				c.ClusterMetadata.ClusterInformation[k] = cluster
+
+	for name, cluster := range c.ClusterMetadata.ClusterInformation {
+		if cluster.RPCName == "" {
+			// filling RPCName with a default value if empty
+			cluster.RPCName = "cadence-frontend"
+		}
+		if name == c.ClusterMetadata.CurrentClusterName {
+			if cluster.RPCAddress == "" {
+				// filling current cluster's RPC address from publicClient if empty
+				cluster.RPCAddress = c.PublicClient.HostPort
```

Comment:
In the linked issue I saw you mentioned that 
> the one in clusterInformation is more generic
> we should deprecate publicClient
so shall we reverse the operation, i.e. automatically fill in `publicClient.HostPort` if not specified?

Reply:
yeah, that's right. Will change to that. 
(side story, there was an issue in HelmChart that the HelmChart maintainer told me it's too difficult to provide a right value in RpcAddress. But they finally figured out this morning Lol. so I think we should be able to fill in publicClient like you said